web/src/app/util/ISOPickers.tsx

```diff
@@ -62,51 +94,54 @@ function useISOPicker(
     setInputValue(value ? dtValue.toFormat(format) : '')
   }, [value, zone])
 
-  const handleChange = (e) => {
-    setInputValue(e.target.value)
-
-    const newVal = inputToISO(e.target.value)
+  const handleChange = (val: Input): void => {
+    const newVal = inputToISO(val)
     // Only fire the parent's `onChange` handler when we have a new valid value,
     // taking care to ensure we ignore any zonal differences.
     if (newVal && newVal !== dtValue.toUTC().toISO()) {
       onChange(newVal)
     }
   }
 
-  // starts with label above textfield so format placeholder can be seen
-  const shrinkInputLabel = (p) => ({
-    ...(p?.InputLabelProps ?? {}),
-    shrink: true,
-  })
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    const val = e.target.value
+    setInputValue(val)
+    handleChange(val)
+  }
 
   if (native) {
     return (
       <TextField
         type={type}
         value={inputValue}
-        onChange={handleChange}
+        onChange={handleInputChange}
+        inputProps={{ min, max }}
         {...otherProps}
-        InputLabelProps={shrinkInputLabel(otherProps)}
+        InputLabelProps={{
+          ...otherProps,
+          shrink: true,
+        }}
       />
     )
   }
 
-  const extraProps = {}
+  const cypressProps: { [key: string]: { 'data-cy'?: string } } = {}
+  cypressProps.DialogProps = { 'data-cy': 'picker-fallback' }
   if (type !== 'time') {
-    extraProps.leftArrowButtonProps = { 'data-cy': 'month-back' }
-    extraProps.rightArrowButtonProps = { 'data-cy': 'month-next' }
+    cypressProps.leftArrowButtonProps = { 'data-cy': 'month-back' }
+    cypressProps.rightArrowButtonProps = { 'data-cy': 'month-next' }
   }
```

Comment:
Maybe call these `testingProps`? without context it would make it more clear to a new dev that may not be familiar with cypress